icu4c/source/test/fuzzer/ucasemap_fuzzer.cc

```diff
@@ -16,23 +16,43 @@ using deleted_unique_ptr = std::unique_ptr<T,std::function<void(T*)>>;
 
 extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
   UErrorCode status = U_ZERO_ERROR;
+  uint8_t rnd8 = 0;
+  uint16_t rnd16 = 0;
+  uint32_t rnd32 = 0;
 
-  auto rng = CreateRng(data, size);
-  const icu::Locale& locale = GetRandomLocale(&rng);
-  uint32_t open_flags = static_cast<uint32_t>(rng());
+  if (size < 7) {
+    return 0;
+  }
+  // Extract one, two, and four bytes from fuzzer data for random selection
+  // purposes.
+  rnd8 = *data;
+  data++;
+  rnd16 = *(reinterpret_cast<const uint16_t *>(data));
+  data = data + 2;
+  rnd32 = *(reinterpret_cast<const uint32_t *>(data));
+  data = data + 4;
+  size = size - 7;
+
+  size_t unistr_size = size/2;
+  std::unique_ptr<char16_t[]> fuzzbuff(new char16_t[size/2]);
+  std::memcpy(fuzzbuff.get(), data, unistr_size * 2);
+
+  const icu::Locale& locale = GetRandomLocale(rnd16);
+  uint32_t open_flags = rnd32;
 
   deleted_unique_ptr<UCaseMap> csm(
       ucasemap_open(locale.getName(), open_flags, &status),
       [](UCaseMap* map) { ucasemap_close(map); });
 
-  if (U_FAILURE(status))
+  if (U_FAILURE(status)) {
     return 0;
+  }
 
   int32_t dst_size = size * 2;
   std::unique_ptr<char[]> dst(new char[dst_size]);
-  auto src = reinterpret_cast<const char*>(data);
+  auto src = reinterpret_cast<const char*>(fuzzbuff.get());
```

Comment:
Might as well have fuzzbuff be <char []> instead of <char16_t[]>, then no casting would be needed. It's only ever used as bytes.

Reply:
Done.